### PR TITLE
Refactor code to implement the strategy design pattern and do some general cleanup

### PR DESCRIPTION
**Pull request by hknehans**
I think this is a pretty good start to the reorganization. I left quite a few TODO comments for myself or others. One in particular is the TODO to reorganize the imports. I need to figure out a good way to do that because I've found importing in Python to be a little bit weird.
Also, you'll see a lot of the data-set files deleted. That's because somehow there were duplicates created.
And I assumed test.py was safe to delete because that looked like old stuff which we know what is going on now.

### CODE REVIEW

File: analysis/Sentiment_Analysis.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        the padded sequences as a 2D array of integers
+    ===================================================================
+    """
+    @classmethod
+    def pre_process(cls, responses, max_len):
+
+        token_sequences = []
+        for response in responses:
+            word_seq = text_to_word_sequence(response)
+            token_seq = [cls.word_index[word] if word in cls.word_index else -1 for word in word_seq]
```
Comment (hknehans) [inline analysis/Sentiment_Analysis.py:30]: I took this code from Drew's model file, but I changed the else here to be a -1 rather than a 0 because the integer 0 is used for padding.

File: analysis/Sentiment_Analysis.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        and does any padding or truncating if necessary
+    Paramaters:
+        responses: an array of strings to be pre-processed
+        max_len: the max word length a response can be
+    Returns:
+        the padded sequences as a 2D array of integers
+    ===================================================================
+    """
+    @classmethod
+    def pre_process(cls, responses, max_len):
```
Comment (hknehans) [inline analysis/Sentiment_Analysis.py:25]: The cls argument here and in the other class methods are required and ties it to the class itself rather than having to create an instance. Similar to a static method in Java.

File: analysis/Thematic_Anlaysis.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# TODO organize imports in separate files
+from analysis.Analysis import Analysis
+from nltk.corpus import stopwords
+from textblob import Word, TextBlob
+from sklearn.decomposition import NMF
+from sklearn.feature_extraction.text import TfidfVectorizer
+import nltk
+
+class Thematic_Analysis(Analysis):
```
Comment (hknehans) [inline analysis/Thematic_Anlaysis.py:9]: We will probably have to do some adjusting to the code in these methods. Specifically the pre_process() method to take the user input the same way it's being taken and pre-processed for the sentiment analysis. Or it could go the other way around. I just think both should read the user's responses the same way.

Also, let's keep us-04-thamtic-model/thematicAnalysisMain.py until we know we have everything from there needed.

File: analysis/analysis.py [hunk 4]
```diff
@@ -1,120 +1,15 @@
-"""
```
Comment (hknehans) [inline analysis/analysis.py:1]: This file was stripped and made into the parent Analysis class for the two child classes. The functionalities are put in those child classes.

File: user_input/input.py [hunk 5]
```diff
@@ -9,6 +9,7 @@
 =================================================
 """
```
Comment (hknehans) [inline user_input/input.py:11]: Should I just go ahead and encapsulate these functions into an Input class?

Comment (bentlee108) [reply user_input/input.py:11]: I'd say if you feel like it would benefit the project, go for it.

**Review by bentlee108 — APPROVED** [review]
